### DISCUSSION TITLE
Is is safe to edit a KafkaTopic that was imported after creation?

### DISCUSSION THREAD

**Discussion by jslusher** [strimzi/strimzi-kafka-operator#9888] [Q&A]
We're using Strimzi operator `0.36.0` at the moment. We use `KafkaConnector` resources but we do not specify `KafkaTopic` resources. I noticed though that for our topics there is a `KafkaTopic` resource associated that has some basic configuration in it. I also discovered that I can edit the `KafkaTopic` and that at least `spec.config` items can be edited and they seem to get set on the topic itself. I have two topics that I want to edit `delete.retention.ms` from their broker default of 24h. Is this safe to do? Is there anything else I should be accounting for when editing these topics in this way or is this exactly like editing the topic directly?

**Comment by scholzj** [comment]
You can do that. But be careful so that the client using the topic doesn't change it back. Also, keep in mind that only the old Bidirectional Topic Operator creates the KafkaTopic resources for topics created in Kafka. The new Unidirectional Topic Operators does not do that.

**Reply by jslusher** [reply]
In the case of clients, would that be the Debezium connector? I've set `topic.creation` settings on it in the `KafkaConnector` `spec.config` to reflect the updated setting so that if we recreate the topic at any point, the setting will be the same. 

What version starts using the Unidirectional Topic Operator? That would mean in the future any topics created by a `KafkaConnector` resource would not also populate with `KafkaTopic`s?

**Reply by scholzj** [reply]
> In the case of clients, would that be the Debezium connector? I've set `topic.creation` settings on it in the `KafkaConnector` `spec.config` to reflect the updated setting so that if we recreate the topic at any point, the setting will be the same.

Depending on the topic, I guess it would be either Connect who creates it for Debezium. Or Debezium itself - I think it creates some of the topics directly. But not sure how exactly they are managed once created to be honest.

> What version starts using the Unidirectional Topic Operator? That would mean in the future any topics created by a `KafkaConnector` resource would not also populate with `KafkaTopic`s?

The Unidirectional Topic OPerator is enabled by default from 0.39 and will be the only option from 0.41. When it is used, you have to create the KafkaTopic resource yourself.

**Reply by scholzj** [reply]
You can also check https://strimzi.io/blog/2023/11/02/unidirectional-topic-operator/ or the docs.

**Reply by jslusher** [reply]
I'll have a look. Thanks!
